src/pages/TCY/components/Stake/StakeInput.tsx

```diff
@@ -80,42 +138,82 @@ export const StakeInput: React.FC<TCYRouteProps> = ({ headerComponent }) => {
+  }, [accountId, setValue])
+
   return (
     <Stack>
       {headerComponent}
-      <TradeAssetInput
-        assetId={selectedStakingAsset?.assetId ?? ''}
-        assetSymbol={selectedStakingAsset?.symbol ?? ''}
-        assetIcon={selectedStakingAsset?.icon ?? ''}
-        onAccountIdChange={noop}
-        label={translate('TCY.stakeInput.amount')}
-        isAccountSelectionDisabled
-        placeholder={translate('TCY.stakeInput.amountPlaceholder')}
-        onToggleIsFiat={setIsFiat}
-        onChange={handleAmountChange}
-        isFiat={isFiat}
-        cryptoAmount={cryptoAmount}
-        fiatAmount={fiatAmount}
-        percentOptions={percentOptions}
-        formControlProps={formControlProps}
-        rightComponent={ReadOnlyAsset}
-      />
-      <CardFooter
-        flexDirection='column'
-        gap={4}
-        bg='background.surface.raised.base'
-        borderBottomRadius='xl'
-      >
-        <Row fontSize='sm' Tooltipbody={tooltipBody}>
-          <Row.Label>{translate('TCY.stakeInput.networkFee')}</Row.Label>
-          <Row.Value>
-            <Amount.Fiat value={0} />
-          </Row.Value>
-        </Row>
-        <Button colorScheme='blue' size='lg' width='full' onClick={handleStake}>
-          {translate('TCY.stakeInput.stake')}
-        </Button>
-      </CardFooter>
+      <FormControl isInvalid={Boolean(errors.amountCryptoPrecision)}>
+        <TradeAssetInput
+          assetId={selectedStakingAsset?.assetId ?? ''}
+          assetSymbol={selectedStakingAsset?.symbol ?? ''}
+          assetIcon={selectedStakingAsset?.icon ?? ''}
+          onAccountIdChange={noop}
+          label={translate('TCY.stakeInput.amount')}
+          isAccountSelectionDisabled
+          placeholder={translate('TCY.stakeInput.amountPlaceholder')}
+          onToggleIsFiat={toggleIsFiat}
+          onChange={handleAmountChange}
+          isFiat={fieldName === 'fiatAmount'}
+          cryptoAmount={amountCryptoPrecision}
+          fiatAmount={fiatAmount}
+          percentOptions={percentOptions}
+          formControlProps={formControlProps}
+          rightComponent={ReadOnlyAsset}
+        />
+      </FormControl>
+      <Card>
+        <CardFooter
+          flexDirection='column'
+          gap={4}
+          bg='background.surface.raised.base'
+          borderBottomRadius='xl'
+        >
+          <Row fontSize='sm' Tooltipbody={tooltipBody}>
+            <Row.Label>{translate('TCY.stakeInput.networkFee')}</Row.Label>
+            <Row.Value>
+              <Skeleton isLoaded={!!estimatedFeesData}>
+                <Amount.Fiat value={estimatedFeesData?.txFeeFiat ?? 0} />
+              </Skeleton>
+            </Row.Value>
+          </Row>
+          <Button
+            colorScheme={isValid ? 'blue' : 'red'}
+            size='lg'
+            width='full'
+            onClick={handleStake}
+            isDisabled={isDisabled}
+            isLoading={isEstimatedFeesDataLoading}
+          >
+            {confirmCopy}
```

Comment:
Non-blocking for this PR, but we probably want to handle no wallet connected here:

<img width="774" alt="Screenshot 2025-05-05 at 6 03 22 pm" src="https://github.com/user-attachments/assets/ee601a61-d7cc-4c15-ac51-2961dd3d7782" />

Network fee fails, no "Connect wallet" copy.

Reply:
Wallet bits tackled in https://github.com/shapeshift/web/pull/9468 - not sure how we'll want to handle the network fee part just yet, it looks at we may be able to hardcode the RUNE fee which would make things even easier here!